pkg/registry/chains/memory/server.go

```diff
@@ -40,9 +40,9 @@ import (
 func NewServer(ctx context.Context, expiryDuration time.Duration, proxyRegistryURL *url.URL, options ...grpc.DialOption) registryserver.Registry {
 	nseChain := chain.NewNetworkServiceEndpointRegistryServer(
 		serialize.NewNetworkServiceEndpointRegistryServer(),
+		setid.NewNetworkServiceEndpointRegistryServer(),
```

Comment:
Why we need setid here?

Reply:
It is a chain for the memory registry. So we need to check for the collisions in registry.

Reply:
maybe setid is a wrong name in this case?

Reply:
Extracted server + error to `checkid` package.